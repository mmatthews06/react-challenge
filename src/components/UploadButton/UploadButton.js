import React from 'react';
import PropTypes from 'prop-types';
import Papa from 'papaparse';
import Button from '@material-ui/core/Button';

export default function UploadButton({ setSamples }) {
  function parseCSV(event) {
    const newFile = event.target.files[0];
    Papa.parse(newFile, {
      header: true,
      dynamicTyping: true,
      delimiter: ',',
      complete: (results) => {
        setSamples(results.data);
      },
    });
  }

  return (
    <label htmlFor="contained-button-file" id="upload-samples">
      <input
        accept=".csv"
        id="contained-button-file"
        type="file"
        hidden
        value=""
        onChange={parseCSV}
      />
      <Button variant="contained" component="span" color="primary">
        Upload Samples
      </Button>
    </label>
  );
}

UploadButton.propTypes = {
  setSamples: PropTypes.func.isRequired,
};
