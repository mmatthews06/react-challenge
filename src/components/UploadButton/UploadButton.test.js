/* global describe, expect, it, jest, File */
import React from 'react';
import { mount, shallow } from 'enzyme';

import UploadButton from './UploadButton';

describe('UploadButton', () => {
  it('should allow you to upload a file', async (done) => {
    expect.assertions(2);
    const headers = 'foo,bar,baz';
    const row1 = '1,2,3';
    const row2 = '5,6,7';
    const fileText = `${headers}\n${row1}\n${row2}`;

    const mockSetSamples = jest.fn(async (samples) => {
      await expect(samples).toHaveLength(2);
      await expect(samples[0]).toEqual({ foo: 1, bar: 2, baz: 3 });
      done();
    });

    const file = new File([fileText], 'foo.csv');
    const input = mount(<UploadButton setSamples={mockSetSamples} />).find('input');
    input.simulate('change', { target: { files: [file] } });
  });

  it('should match a snapshot', () => {
    expect(shallow(<UploadButton />)).toMatchSnapshot();
  });
});