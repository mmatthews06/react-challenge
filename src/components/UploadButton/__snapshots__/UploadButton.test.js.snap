// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadButton should match a snapshot 1`] = `
<label
  htmlFor="contained-button-file"
  id="upload-samples"
>
  <input
    accept=".csv"
    hidden={true}
    id="contained-button-file"
    onChange={[Function]}
    type="file"
    value=""
  />
  <WithStyles(ForwardRef(Button))
    color="primary"
    component="span"
    variant="contained"
  >
    Upload Samples
  </WithStyles(ForwardRef(Button))>
</label>
`;
